bigquery_etl/cli/metadata.py

```diff
@@ -0,0 +1,87 @@
+"""bigquery-etl CLI metadata command."""
+from pathlib import Path
+from typing import Optional
+
+import click
+
+from bigquery_etl.metadata.parse_metadata import DatasetMetadata, Metadata
+
+from ..cli.utils import paths_matching_name_pattern, project_id_option, sql_dir_option
+
+
+@click.group(
+    help="""
+        Commands for managing bqetl metadata.
+        """
+)
+@click.pass_context
+def metadata(ctx):
+    """Create the CLI group for the metadata command."""
+    pass
+
+
+@metadata.command(
+    help="""
+    Update metadata yaml files.
+    Updates workgroup access metadata based on the dataset_metadata.yaml and
+    deprecation metadata.
+
+    Example:
+     ./bqetl metadata update ga_derived.downloads_with_attribution_v2
+    """,
+    context_settings=dict(
+        ignore_unknown_options=True,
+        allow_extra_args=True,
+    ),
+)
+@click.argument("name")
+@project_id_option()
+@sql_dir_option
+def update(name: str, sql_dir: Optional[str], project_id: Optional[str]) -> None:
+    """Update metadata yaml file."""
+    table_metadata_files = paths_matching_name_pattern(
+        name, sql_dir, project_id=project_id, files=["metadata.yaml"]
+    )
+    dataset_metadata_path = None
+    # create and populate the dataset metadata yaml file if it does not exist
+    for table_metadata_file in table_metadata_files:
+        dataset_metadata_path = (
+            Path(table_metadata_file).parent.parent / "dataset_metadata.yaml"
+        )
+        if not dataset_metadata_path.exists():
```

Comment:
@scholtzan please let me know your thoughts on this. I almost can't remember why we added the `if not dataset_metadata_path.exists():`

Reply:
There are some datasets that don't have `dataset_metadata.yaml`: https://github.com/mozilla/bigquery-etl/tree/main/sql/moz-fx-data-shared-prod/messaging_system_derived

Reply:
We should probably add them though, so that we can assume a `dataset_metadata.yaml` exists

Reply:
I meant by `i.e. it is already true that this file exists in generated branches` that the file gets generated at some point: https://github.com/mozilla/bigquery-etl/blob/generated-sql/sql/moz-fx-data-shared-prod/messaging_system_derived/dataset_metadata.yaml. So my assumption is that we already have the generation logic somewhere for this and have had it for some time.

Reply:
I did notice when double checking this that https://github.com/mozilla/bigquery-etl/tree/generated-sql/sql/moz-fx-data-shared-prod/udf_js_lib doesn't get a `dataset_metadata.yaml` file, but that appears to be because this isn't a dataset. At any rate, `dataset_metadata.yaml` is a de facto requirement for a dataset to be generated from bqetl metadata and requiring its existence de jure would be reasonable.